201-representing-CNAB-in-OCI.md

```diff
@@ -0,0 +1,132 @@
+  ],
+  "images": null,
+  "parameters": null,
+  "credentials": null
+}
+```
+
+The `bundle.json` is serialized into canonical JSON and is pushed as a blob to the registry. This blob is then referenced by its digest (for example `sha256:e91b9dfcbbb3b88bac94726f276b89de46e4460b55f6e6d6f876e666b150ec5b`) as the object config.
+
+The bundle is encapsulated in an OCI manifest which is then referenced as part of an OCI index (or manifest list):
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.cnab.bundle.config.v1+json",
+    "digest": "sha256:e91b9dfcbbb3b88bac94726f276b89de46e4460b55f6e6d6f876e666b150ec5b",
+    "size": 498
+  },
+  "layers": null
+}
+```
+
+The bundle file MUST be stored in the OCI registry, as it is the canonical representation of a CNAB bundle, and is the object that is used to compute the signature for a thin bundle.
+While implementations MAY choose to surface information from the bundle as top-level annotations in the OCI representations, they MUST store the unmodified bundle file as a blob.
+
+The bundle file and registry location MAY be used by implementations to generate relocation maps at runtime.
+
+### Invocation images
+
+> In practice most images are built as Docker v2 images so we should allow these
+
+Each invocation image SHOULD be packaged as an OCI image but MAY be packaged as a Docker v2 image. If there are multiple invocation images, these SHOULD be referenced by an OCI index but MAY be referenced by a Docker manifest list.
+
+### Component images
+
+> In practice most images are built as Docker v2 images so we should allow these
+
+Each component image SHOULD be packaged as an OCI image but MAY be packaged as a Docker v2 image. In the case that a component has multiple equivalent forms, such as a multiarch container image, each component MUST be packaged and then SHOULD be referenced by an OCI index but MAY be referenced by a Docker manifest list.
+
+## Top-level representation
+
+The top-level representation of the CNAB SHOULD be an OCI index but may be a Docker manifest list.
+
+The `manifests` list MUST be filled in the following order:
+
+- A reference to packaged `bundle.json` manifest
+- References to the invocation images
+- References to the component images
+
+### Example
+
+```json
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:6ec4fd695cace0e3d4305838fdf9fcd646798d3fea42b3abb28c117f903a6a5f",
+      "size": 188,
+      "annotations": {
+        "io.cnab.manifest.type": "config"
+      }
+    },
+    {
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+      "digest": "sha256:a59a4e74d9cc89e4e75dfb2cc7ea5c108e4236ba6231b53081a9e2506d1197b6",
+      "size": 942,
+      "annotations": {
+        "io.cnab.manifest.type": "invocation"
+      }
+    }
+  ],
+  "annotations": {
+    "io.cnab.keywords": "[\"helloworld\",\"cnab\",\"tutorial\"]",
```

Comment:
Should the spec define how to populate or use the annotations?

Reply:
Opened #382 to track this.